### PR TITLE
AS-106 "blog or tweet" expectation

### PR DESCRIPTION
**Pull request by YuriHechter**

### CODE REVIEW

File: consts/consts.go [hunk 1]
```diff
@@ -318,6 +318,9 @@ const ES_RESULT_TYPE_TAGS = "tags"
 const ES_RESULT_TYPE_BLOG_POSTS = "posts"
 const ES_RESULT_TYPE_TWEETS = "tweets"
 
+// Result of many tweets in one hit
+const SEARCH_RESULT_TWEETS_MANY = "tweets_many"
```
Comment (bbfsdev) [inline consts/consts.go]: This is being used in eval only?
Should also be used in serving?

File: cmd/eval.go [hunk 2]
```diff
@@ -209,9 +209,18 @@ func evalFn(cmd *cobra.Command, args []string) {
 		utils.Must(err)
 		printResults(results)
 		printLosses(results, losses)
-		search.WriteResults(reportPath, evalSet, results)
-		_, err = search.WriteResultsByExpectation(flatReportPath, evalSet, results)
-		utils.Must(err)
+		if _, ferr := os.Stat(reportPath); os.IsNotExist(ferr) {
+			log.Warnf("Cannot write results: reportPath ('%s') is not found!", reportPath)
```
Comment (bbfsdev) [inline cmd/eval.go]: Why do you check stat here?
What are you trying to solve?

Comment (bbfsdev) [reply cmd/eval.go]: Why file has to exist?

Comment (bbfsdev) [reply cmd/eval.go]: You are writing the report, it might not exist yet, right?

File: search/quality_eval.go [hunk 3]
```diff
@@ -169,6 +172,7 @@ const (
 	FILTER_NAME_TOPIC        = "topic"
 	FILTER_NAME_CONTENT_TYPE = "contentType"
 	PREFIX_LATEST            = "[latest]"
+	BLOG_OR_TWEET_MARK       = "blog" //  TBD consider to change the value
```
Comment (bbfsdev) [inline search/quality_eval.go]: Change this to blog_or_tweet   // Update the recall set that you are committing.
When we commit this we also update the recall set in spreadsheet.

File: search/quality_eval.go [hunk 4]
```diff
@@ -488,6 +495,12 @@ func HitMatchesExpectation(hit *elastic.SearchHit, hitSource HitSource, e Expect
 	if hitType == "result" {
 		hitType = hitSource.ResultType
 	}
+	if e.Type == ET_BLOG_OR_TWEET {
```
Comment (bbfsdev) [inline search/quality_eval.go]: How did you check it works?

File: consts/consts.go (hunk 1 above)
Comment (YuriHechter) [reply consts/consts.go]: also used in search, see https://github.com/Bnei-Baruch/archive-backend/pull/124

File: search/quality_eval.go (hunk 4 above)
Comment (YuriHechter) [reply search/quality_eval.go]: debug

File: cmd/eval.go (hunk 2 above)
Comment (YuriHechter) [reply cmd/eval.go]: changed to check that this parameters are defined (without the checking we may encounter panic)

File: search/quality_eval.go (hunk 3 above)
Comment (YuriHechter) [reply search/quality_eval.go]: changed, recall updated in source code and spreadsheet

**Comment by bbfsdev** [comment]
Merged manually.

**Pull request closed without merging**